### PR TITLE
Add dialect readout command and docs

### LINKED ISSUE #435: What dialects sqlfluff supports?
I can't find the list of supported dialects in the documentation. I wonder if this supports Oracle for example.

### PR DESCRIPTION
**Pull request by alanmcruickshank**
Fixes #435 .

I couldn't find a more elegant way of using `autodoc` to document the dialects (which would have been nice), so wrote them out longhand for the docs. The cli command is much more well integrated and tidier.

### CODE REVIEW

File: docs/source/dialects.rst [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+dialects. Each inherits from another, up to the root `ansi` dialect.
+
+For a canonical list of supported dialects, run the
+:program:`sqlfluff dialects` command, which will output a list of the
+current dialects available on your installation of sqlfluff.
+
+.. note::
+
+    For technical users looking to add new dialects or add new features
+    to existing ones; the dependent nature of how dialects have been
```
Comment (greg-finley) [inline docs/source/dialects.rst:18]: change `;` to `,`

File: docs/source/dialects.rst [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+ANSI
+----
+
+This is the base dialect which holds most of the definitions of common
+SQL commands and structures. If the dialect which you're actually using
+isn't specifically implemented by sqlfluff, using this dialect is a good
+place to start.
+
+This dialect doesn't intend to be brutal in adhering to (and only to) the
+ANSI SQL spec *(mostly because ANSI charge for access to that spec)*. It aims
```
Comment (greg-finley) [inline docs/source/dialects.rst:50]: I think it should be `ANSI charges` but not sure if this is an Americanism

File: docs/source/dialects.rst [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+The dialect for `Google BigQuery`_.
+
+.. _`Google BigQuery`: https://cloud.google.com/bigquery/
+
+.. _snowflake_dialect_ref:
+
+Snowflake
+---------
+
+The dialect for `Snowflake`_, which has much of it's syntax
```
Comment (greg-finley) [inline docs/source/dialects.rst:102]: `it's` -> `its`

**Review by greg-finley — APPROVED** [review]
Can't really assess the Python changes, but left some copy-editing nits.

**Comment by felipecrs** [comment]
LGTM

**Comment by alanmcruickshank** [comment]
Muchas gracias 🙏

**Pull request merged**